content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,315 @@
+This article is about Sockets. It will give you a glimpse of how clients and servers communicate over the internet.
+
+### Introduction
+
+Socket programming is a means of communicating data between two computers across a network. Connections can be made using either a **connection-oriented protocol** or a **connectionless protocol**. TCP/IP, a connection-oriented protocol, will be used in our case.
+
+Before exchanging data, two computers must establish a link in connection-oriented protocols. In contrast, you don't require every packet to arrive at its intended destination for connectionless. It is made feasible via UDP (User Datagram Protocol).
+
+To demonstrate sockets further, we shall use the Client/Server architecture. Client and server communicate by writing to and reading from the socket connection. The machines must have each other's network information to make this connection.
+
+### Table of contents:
+- [Definition of a Socket](#definition-of-a-socket)
+- [Programming on the Server-side ](#programming-on-the-server-side)
+- [Programming on the Client-side](#programming-on-the-client-side)
+- [How to Run Programs in a Variety of Applications](#how-to-run-programs-in-a-variety-of-applications)
+- [Conclusion](#conclusion)
+
+### Definition of a Socket
+
+A Socket is a communication **endpoint** that serves as a link between two machines on a network. It has a socket number, which the TCP/IP layer can use to identify the application that receives the data. Additionally, a port number and an IP address are what make up an endpoint.
```

Comment:
Change "socket number" to "port number" to be more technically accurate. According to [this source](https://docs.oracle.com/javase/tutorial/networking/sockets/definition.html):

> A socket is one endpoint of a two-way communication link between two programs running on the network. A socket is bound to a port number so that the TCP layer can identify the application that data is destined to be sent to.

Notice how it says "port number" instead of "socket number". The way you phrased it here made me think there was a separate thing called a "socket number" which is different than a port number.